cmd/influx/cli/cli.go

```diff
@@ -139,56 +147,56 @@ func (c *CommandLine) Run() {
 
 	c.Version()
 
-	var historyFile string
+	var historyFilePath string
 	usr, err := user.Current()
-	// Only load history if we can get the user
+	// Only load/write history if we can get the user
 	if err == nil {
-		historyFile = filepath.Join(usr.HomeDir, ".influx_history")
-
-		if f, err := os.Open(historyFile); err == nil {
-			c.Line.ReadHistory(f)
-			f.Close()
+		historyFilePath = filepath.Join(usr.HomeDir, ".influx_history")
+		if c.historyFile, err = os.OpenFile(historyFilePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0640); err == nil {
+			defer c.historyFile.Close()
+			c.Line.ReadHistory(c.historyFile)
 		}
 	}
 
+Loop:
```

Comment:
I actually found about labels in Go by [looking at other pieces of the InfluxDB code](https://github.com/influxdb/influxdb/blob/905437cc5aa179901ece90a99bd991c825129362/influxql/parser.go#L414). And while I never liked labels either, at this time it's a compromise I can live for the time being.